src/components/Popups/PopupCursor.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { FormattedMessage } from 'react-intl';
+import messages from './messages';
+import PopupBase from './PopupBase';
+import { Options, VirtualElement } from './Popper';
+import './PopupCursor.scss';
+
+type Props = {};
+
+const options: Partial<Options> = {
+    modifiers: [
+        {
+            name: 'offset',
+            options: {
+                offset: [0, 20],
+            },
+        },
+    ],
+    placement: 'right',
+};
+
+const generateGetBoundingClientRect = (x = 0, y = 0) => {
+    return () => ({
+        width: 0,
+        height: 0,
+        top: y,
+        right: x,
+        bottom: y,
+        left: x,
+    });
+};
+
+export default function PopupCursor(props: Props): JSX.Element {
+    const popupRef = React.useRef<PopupBase>(null);
```

Comment:
Should this be `<PopupBase | null>` or does TS infer it?

Reply:
> Should this be `<PopupBase | null>` or does TS infer it?

Currently, `popupRef.current` is read-only. If we add ` | null`, the ref will become mutable, which is not necessary in the code.